viam-cartographer.go

```diff
@@ -143,7 +143,10 @@ func initSensorProcess(cancelCtx context.Context, cartoSvc *CartographerService)
 	cartoSvc.sensorProcessWorkers.Add(1)
 	go func() {
 		defer cartoSvc.sensorProcessWorkers.Done()
-		sensorprocess.Start(cancelCtx, spConfig)
+		if jobDone := sensorprocess.Start(cancelCtx, spConfig); jobDone {
+			cartoSvc.jobDone = true
+			cartoSvc.cancelSensorProcessFunc()
```

Comment:
Is there a harm in calling this function here if it is also called in close? Should `cartoSvc.cancelSensorProcessFunc()` only be called in close if jobDone is false?

Reply:
I'd recommend testing reconfiguring after jobdone is set to true to confirm this doesn't affect the Close() function, but assuming it doesn't this looks great.